chapter07/chapter07.adoc

```diff
@@ -0,0 +1,248 @@
+
+Metrics, enables developers and operators to monitor and measure the behavior of microservices at runtime. This observability is crucial for:
+
+- *Performance Tuning*: Identifying bottlenecks and optimizing resource utilization to ensure services are running efficiently.
+- *Scalability*: Making informed decisions on when to scale services up or down based on real-time data on load and performance.
+- *Troubleshooting*: Quickly pinpointing issues by analyzing trends in performance metrics, leading to reduced downtime.
+- *Service Health Monitoring*: Complementing the MicroProfile Health checks by providing deeper insights into the internal state of a service, beyond simple up/down statuses.
+
+=== Types of Metrics
+
+MicroProfile Metrics offers a range of customizable metrics that can be used to measure and monitor microservices' performance. It allows developing microservices that are observable, manageable, and which provide insights into their behavior. 
+
+The MicroProfile Metrics specification includes four different types of metrics that serve specific monitoring purposes: Counter, Gauge, Histogram, and Timer. Each of these types offers unique insights into different aspects of application behavior and performance. Below is the breakdown of available metric types:
+
+1. *Counter*: It is a simple metric type that represents a single numerical value that can only increase over time. This metric is typically used to count occurrences of certain events, such as the number of requests processed, items created, or tasks completed. Monitoring tools like Prometheus are commonly used to analyze changes in the Counter's value over specific intervals. These tools can track the differences in the Counter's value across time periods, providing insights into the rate of occurrences and trends. 
+
+2. *Gauge*: It is a metric type that measures an instantaneous value of something , which can arbitrarily go up or down. It’s used to capture the value of a metric at a particular point in time like the size of a queue, memory usage, or current number of active user sessions. Gauges are typically used for values that change over time, providing a current "gauge" of the system's state..
+
+3. *Histograms*: They provide a distribution of values for a given metric, which are useful for identifying performance outliers. It measures the frequency of values in different ranges (or "buckets") and is useful for tracking the distribution of values, such as response times or data sizes. Histograms can give insights into the average, percentiles, and trends of the measured data over time.
+
+4. *Timer*: It is a specialized metric type that aggregates timing durations and provides data such as the count, total time, mean, and maximum duration. It also usually reports the duration distribution. Timers are invaluable for tracking the duration of certain activities or operations within your application, such as processing time or method execution time.
```

Comment:
It also usually reports
->
It can also report

Reply:
Fixed with a244b0b787c58554fc3eb15950a1efff180cf052